html_css/aria.md

```diff
@@ -0,0 +1,276 @@
+
+* modify an element's appearance,
+* modify an element's behavior,
+* add focusability, or
+* add keyboard event handling.
+
+When you use ARIA, you must take additional steps to add in any missing semantics or functionality. We'll go over this in more detail later in the lesson.
+
+### The Accessibility Tree
+
+Before we get into some of the ARIA attributes available to us, it's important to know some basics of what the accessibility tree is, as it will help you understand just what those attributes are actually doing.
+
+The accessibility tree is based on the DOM, something you should be very familiar with by now. While the DOM represents nodes and objects that make up a web page, the accessibility tree contains only the accessibility related information that will be used by assistive technologies. The way ARIA works is by modifying properties of the objects that make up this accessibility tree. Each of these objects has at least the following four properties:
+
+* **Name**: Also known as the "accessible name", this is what assistive technologies announce to a user and what separates elements of the same type from one another. The name may be set by one or more native labels, including the text contents of an element, the `<label>` element, or the `alt` attribute, to name a few.
+* **Description**: This is what assistive technologies announce in addition to its accessible name.
+* **Role**: This is what an element's purpose is, or what kind of "thing" it is, e.g. a button, a checkbox, a list, etc.
+* **State**: This is a list of any state the element may have, such as whether it's focusable, checked, expanded, etc.
+
+### ARIA Labels
+
+ARIA labels help users of assistive technologies better understand the content on a web page by overriding native labels or providing additional descriptive text. Unlike the `<label>` element, ARIA labels aren't limited to being used on only a select few elements, though they still have their own limitations.
+
+#### `aria-label`
+
+The `aria-label` attribute overrides any native label and modifies the name property in the accessibility tree, though it's best used when an element doesn't already have a native label. When you add `aria-label` to an element, you pass in a string as the value, which will become that elements accessible name. `aria-label` doesn't work on every HTML element, though. Adding the attribute to a plain `<div>` or a `<span>` will have no effect, unless you give those elements a role of some kind.
+
+A common use for this attribute can be for the "close" button often seen in menus or modals:
+
+~~~html
+<button type='button' aria-label='Close modal'>X</button>
+~~~
+
+Instead of a screen reader announcing, "X, button", which wouldn't make any sense to the user, it would announce, "Close modal, button". The same thing can be done for the hamburger icon commonly used for mobile or hidden menus. Another way you could use `aria-label` is on landmark elements:
+
```

Comment:
Did you mean mention the purpose of `id` in the opening paragraph, before the `aria-label` subsection? I do mention using the `id` for the `aria-labelledby` attribute in the subsection that follows, so just want to make sure if that's what you meant.

Reply:
I just think we need more information about when to use `id`. In the earlier courses we imply to not use `id` for nearly anything. If we're shaking it up for them here, it makes sense to explain when they should or should not use it. `id` is often abused, so some guidelines to how it relates to A11y would be great.

Reply:
Maybe something like the following could be added to the opening paragraph? Might be a bit verbose, though.

"You may remember being warned against using the `id` attribute. Normally, you want to avoid overusing it or even using it when it isn't necessary (which more often than not, it isn't, and you should stick with using classes). There are several ARIA attributes, however, that actually *require* another element to have an `id`. 

When using such ARIA attributes, you would provide an `id` to one element, and you would then pass in that `id` value as another element's ARIA attribute value. This creates a link between the elements, similar to how a `<label>` element's `for` attribute creates a link to an input's `id`. Of the three types of ARIA labels mentioned below, both `aria-labelledby` and 'aria-describedby` are two of the ARIA attributes that require another element to be given an `id`."

Reply:
Love it, great work!